### PR TITLE
Move graphql tools web

### PR DESCRIPTION
**Pull request by michenly**
## Description

This PR move all the packages over from https://github.com/Shopify/graphql-tools-web. 
Will be publishing a minor version of all 7 packages added indicating the move.

🗒️ Note that while the CLA status is failing. It is failing due to the contribution of an ex-employee in https://github.com/Shopify/graphql-tools-web

The review only need to be done on the [last commit](https://github.com/Shopify/quilt/pull/1755/commits/854a10090dbcb6fa207fe5245d027407491bc58b), the rest of history are from graphql-tools-web.

## Tophat

Test using sewing-kit https://github.com/Shopify/sewing-kit/pull/2599

1. graphql-config-utilities
2. graphql-mini-transforms
3. graphql-typed
4. graphql-typescript-definitions
5. graphql-validate-fixtures
6. graphql-tool-utilities

Test using web (along side sk beta) https://github.com/Shopify/web/pull/39762

1. graphql-fixtures
2. @shopify/graphql-testing@4.4.1-beta-3
3. @shopify/jest-mock-apollo@5.1.2-beta-3
4. @shopify/react-graphql-universal-provider@3.6.1-beta-3
5. @shopify/react-graphql@6.3.1-beta-3

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

**Comment by michenly** [comment]
🗒️ Note that while the CLA status is failing. It is failing due to the contribution of an ex-employee in https://github.com/Shopify/graphql-tools-web

File: .github/workflows/node-ci.yml [hunk 1]
```diff
@@ -185,7 +185,7 @@ jobs:
         with:
           path: |
             .sewing-kit/
-            ./packages/*/build/
+            ./packages/*/build1/
```
Comment (michenly) [inline .github/workflows/node-ci.yml:188]: mental note to switch this back when I figure out how to clear cache properly

https://github.com/actions/cache/issues/2#issuecomment-673493515

File: .github/workflows/node-ci.yml [hunk 2]
```diff
@@ -186,9 +186,9 @@ jobs:
           path: |
             .sewing-kit/
             ./packages/*/build/
-          key: ${{ runner.os }}-build-${{ github.sha }}
+          key: ${{ runner.os }}-build-${{ secrets.CACHE_VERSION }}-${{ github.sha }}
           restore-keys: |
-            ${{ runner.os }}-build-
+            ${{ runner.os }}-build-${{ secrets.CACHE_VERSION }}-
```
Comment (michenly) [inline .github/workflows/node-ci.yml:191]: Add a version using GitHub repo environment variable. We can use this to do cache busting going forward

**Comment by michenly** [comment]
Sewing-kit test using beta release here https://github.com/Shopify/sewing-kit/pull/2599

**Review by vsumner — APPROVED** [review]

**Review by vsumner — APPROVED** [review]

**Pull request merged**